l10n_br_zip/models/l10n_br_zip.py

```diff
@@ -113,25 +121,64 @@ def zip_search(self, obj):
             result = self.set_result(zip_ids[0])
             obj.write(result)
             return True
-        else:
-            if len(zip_ids) > 1:
-                obj_zip_result = self.env['l10n_br.zip.result']
-                zip_ids = obj_zip_result.map_to_zip_result(
-                    zip_ids, obj._name, obj.id)
-
-                return self.create_wizard(
-                    obj._name,
-                    obj.id,
-                    country_id=obj.country_id.id,
-                    state_id=obj.state_id.id,
-                    city_id=obj.city_id.id,
-                    district=obj.district,
-                    street=obj.street,
-                    zip_code=obj.zip,
-                    zip_ids=[zip.id for zip in zip_ids]
-                )
-            else:
-                raise UserError(_('Nenhum registro encontrado'))
+
+        elif len(zip_ids) > 1:
+
+            obj_zip_result = self.env['l10n_br.zip.result']
+            zip_ids = obj_zip_result.map_to_zip_result(
+                zip_ids, obj._name, obj.id)
+
+            return self.create_wizard(
+                obj._name,
+                obj.id,
+                country_id=obj.country_id.id,
+                state_id=obj.state_id.id,
+                city_id=obj.city_id.id,
+                district=obj.district,
+                street=obj.street,
+                zip_code=obj.zip,
+                zip_ids=[zip.id for zip in zip_ids]
+            )
+        elif not zip_ids:
+
+            zip_str = misc.punctuation_rm(obj.zip)
+            try:
+                res = pycep_correios.consultar_cep(zip_str)
+
+                # Search Brazil id
+                country = self.env['res.country'].search(
+                    [('code', '=', 'BR')], limit=1)
+
+                # Search state with state_code and country id
+                state = self.env['res.country.state'].search([
+                    ('code', '=', res['uf']),
+                    ('country_id', '=', country.id)], limit=1)
+
+                # search city with name and state
+                city = self.env['res.city'].search([
+                    ('name', '=', res['cidade']),
+                    ('state_id.id', '=', state.id)], limit=1)
+
+                values = {
+                    'zip': zip_str,
+                    'street': res['end'],
+                    'district': res['bairro'],
+                    'city_id': city.id or False,
+                    'state_id': state.id or False,
+                    'country_id': country.id or False,
+                }
+
+                # Create zip object
+                self.env['l10n_br.zip'].create(values)
+
+                zip_ids = self.search(domain)
+                result = self.set_result(zip_ids[0])
+                obj.write(result)
+                return True
+
+            except Exception as e:
```

Comment:
This is a big exception, you may want to try first the exceptions and not use this too general one.